heidrun/ct_dc.rb

```diff
@@ -0,0 +1,95 @@
+
+# If more than one dc:contributor property exists then it drops the last one
+# If one or zero exist return nil
+contributor_top = lambda do |r|
+  contrib = r['dc:contributor']
+  count = r['dc:contributor'].count
+  (count > 1) ? contrib.first_value(count - 1) : contrib.last_value(0)
+end
+
+Krikri::Mapper.define(:ct_dc,
+                      :parser => Krikri::OaiDcParser,
+                      :parser_args => '//rdf:RDF/rdf:Description') do
+  provider :class => DPLA::MAP::Agent do
+    uri 'http://dp.la/api/contributor/connecticut'
+    label 'Connecticut Digital Library'
+  end
+
+  dataProvider :class => DPLA::MAP::Agent do
+    providedLabel record.field('dc:contributor').last_value
+  end
+
+  isShownAt :class => DPLA::MAP::WebResource do
+    uri record.field('dc:identifier')
+  end
+
+  preview :class => DPLA::MAP::WebResource do
+    uri record.field('dc:source')
+  end
+
+  originalRecord :class => DPLA::MAP::WebResource do
+    uri record_uri
+  end
+
+  sourceResource :class => DPLA::MAP::SourceResource do
+    collection  :class => DPLA::MAP::Collection,
+                :each => record.field('xmlns:header', 'xmlns:setSpec'),
+                :as => :coll do
+      title coll
+    end
+
+    contributor :class => DPLA::MAP::Agent,
+                :each => record.map( &contributor_top ).flatten,
```

Comment:
@no-reply I tried this yesterday and a second look today but couldn't get it to click. 

Passing a regular Ruby array causes this to choke here. I've tried remapping the array to a ValueArray without success.
https://github.com/dpla/KriKri/blob/develop/lib/krikri/mapping_dsl/parser_methods.rb#L107

 I'm tempted to revert back to the lambda method and open an issue to fix this when https://github.com/dpla/KriKri/pull/240 is merged and released. Thoughts?